src/components/Header/Header.tsx

```diff
@@ -55,18 +57,32 @@ export const Header = chakra(() => {
     return () => window.removeEventListener("keydown", close);
   }, []);
 
+  const loginDataCallback = (data) => {
+    // If the statusCode of the returned data is 401 and the expired
+    // key is set to true, try to refresh the accessToken.
+    if (data.data.statusCode === 401 && data.data.expired === true) {
+      refreshAccessToken(
+        tokenRefreshLink,
+        loginDataCallback,
+        deleteCookieValue
+      );
+      // Else, extract the patron's name from the returned data.
+    } else {
+      const fullName = extractPatronName(data);
+      setPatronName(fullName);
+    }
+  };
+
   useEffect(() => {
-    const { cookieValue, accessToken } = getCookieValue();
+    // Aftering mounting,look for a cookie named "nyplIdentityPatron"
+    // and try to grab its value.
+    const { accessToken, cookieValue } = getCookieValue();
+    // If the cookie exists, use its accessToken to make a fetch
+    // request for the patron's data.
     if (cookieValue) {
-      if (!patronDataReceived) {
-        fetchPatronData(accessToken, (data) => {
-          const fullName = extractPatronName(data);
-          setPatronName(fullName);
-          setPatronDataReceived(true);
-        });
-      }
+      getLoginData(accessToken, loginDataCallback);
     }
-  }, [patronDataReceived]);
+  }, []);
```

Comment:
The linter complains here but the IDE's linter does not. Add the appropriate eslint ignore comment here.